FoundationsMastery/CustomString.cs

```diff
@@ -0,0 +1,98 @@
+﻿using FoundationsMastery.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FoundationsMastery
+{
+    public class CustomString : ICustomString
+
+    {
+        public IEnumerable<char> Contents { get; set; }// this will have value from the constructor
+
+        public CustomString(IEnumerable<char> contents)
+        {
+            this.Contents = contents; // sets the value of the contents property from what is passed in the construtor
+        }
+
+        public CustomString()
+        {
+            this.Contents = new List<char>();//Always set empty constructor to a new list
+        }
+
+
+        public CustomString(char[] mychars)
+        {
+            Contents = mychars;
+        }
+
+        public int Length // give the length of the contents IEnumerable
+        {
+            get
+            {
+                return Contents.Count();
+            }
+        }
+
+        public void Clear()
+        {
+            Contents = null;
```

Comment:
But what if we access `Length` after calling `Clear`? `NullReferenceException`!!! Couple ways to safeguard against this